internal/service/vmservice/vm.go

```diff
@@ -257,21 +259,70 @@ func getMachineAddresses(scope *scope.MachineScope) ([]clusterv1.MachineAddress,
 		},
 	}
 
+	networkSpec := scope.ProxmoxMachine.Spec.Network
+	if networkSpec == nil {
+		networkSpec = &infrav1alpha1.NetworkSpec{
+			Default: &infrav1alpha1.NetworkDevice{},
+		}
+	}
+	var ipv4, ipv6 string
+	// get ip addresses from vm if dhcp is enabled
+	if hasDHCPEnabled(scope.InfraCluster.ProxmoxCluster.Spec.IPv4Config,
+		networkSpec.Default,
+		infrav1alpha1.DefaultNetworkDevice,
+		infrav1alpha1.IPV4Format) ||
+		hasDHCPEnabled(scope.InfraCluster.ProxmoxCluster.Spec.IPv6Config,
+			networkSpec.Default,
+			infrav1alpha1.DefaultNetworkDevice,
+			infrav1alpha1.IPV6Format) {
```

Comment:
It would actually be less code if you move the default interface handling out of `hasDHCPEnabled` and use a few intermediate variables, e.g.

```go
if dev := networkSpec.Default; scope.InfraCluster.ProxmoxCluster.Spec.DHCPEnabled() || (dev != nil && dev.DHCP4 || dev.DHCP6) {
```

But more importantly, why is the default device DHCP setting even considered when DHCP is enabled on the cluster?